pkg/collector/listeners/kubelet.go

```diff
@@ -0,0 +1,233 @@
+}
+
+func NewKubeletListener() (ServiceListener, error) {
+	watcher, err := kubelet.NewPodWatcher(5 * time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to kubelet, Kubernetes listener will not work: %s", err)
+	}
+	return &KubeletListener{
+		watcher:  watcher,
+		services: make(map[ID]Service),
+		ticker:   time.NewTicker(5 * time.Second),
+		stop:     make(chan bool),
+	}, nil
+}
+
+func (l *KubeletListener) Listen(newSvc chan<- Service, delSvc chan<- Service) {
+	// setup the I/O channels
+	l.newService = newSvc
+	l.delService = delSvc
+
+	go func() {
+		for {
+			select {
+			case <-l.stop:
+				return
+			case <-l.ticker.C:
+				// Compute new/updated pods
+				updatedPods, err := l.watcher.PullChanges()
+				if err != nil {
+					log.Error(err)
+					continue
+				}
+				for _, pod := range updatedPods {
+					// Ignore pending/failed/succeeded/unknown states
+					if pod.Status.Phase == "Running" {
+						l.processNewPod(pod)
+					}
+				}
+				// Compute deleted pods
+				expiredContainerList, err := l.watcher.ExpireContainers()
+				if err != nil {
+					log.Error(err)
+					continue
+				}
+				for _, containerID := range expiredContainerList {
+					l.removeService(ID(containerID))
+				}
+			}
+		}
+	}()
+}
+
+func (l *KubeletListener) Stop() {
+	l.ticker.Stop()
+	l.stop <- true
+}
+
+func (l *KubeletListener) processNewPod(pod *kubelet.Pod) {
+	for _, container := range pod.Status.Containers {
+		l.createService(ID(container.ID), pod)
+	}
+}
+
+func (l *KubeletListener) createService(id ID, pod *kubelet.Pod) {
+	svc := PodContainerService{
+		ID:       id,
+		PodInfos: pod,
+	}
+
+	podName := pod.Metadata.Name
+	_, err := svc.GetADIdentifiers()
+	if err != nil {
+		log.Errorf("Failed to get AD identifiers for pod %s: %s", podName, err)
+	}
+	_, err = svc.GetHosts()
+	if err != nil {
+		log.Errorf("Failed to get hosts for pod %s: %s", podName, err)
+	}
+	_, err = svc.GetPorts()
+	if err != nil {
+		log.Errorf("Failed to get ports for pod %s: %s", podName, err)
+	}
+	_, err = svc.GetTags()
+	if err != nil {
+		log.Errorf("Failed to get tags for pod %s: %s", podName, err)
+	}
+
+	l.m.Lock()
+	l.services[ID(id)] = &svc
+	l.m.Unlock()
+
+	l.newService <- &svc
+}
+
+func (l *KubeletListener) removeService(cID ID) {
+	l.m.RLock()
+	svc, ok := l.services[cID]
+	l.m.RUnlock()
+
+	if ok {
+		l.m.Lock()
+		delete(l.services, cID)
+		l.m.Unlock()
+
+		l.delService <- svc
+	} else {
+		log.Debugf("Container %s not found, not removing", cID)
+	}
+}
+
+// GetID returns the service ID
+func (s *PodContainerService) GetID() ID {
+	return s.ID
+}
+
+// GetADIdentifiers returns the service AD identifiers
+func (s *PodContainerService) GetADIdentifiers() ([]string, error) {
+	if len(s.ADIdentifiers) == 0 {
+		searchedId := string(s.ID)
+		for _, container := range s.PodInfos.Status.Containers {
+			if container.ID == searchedId {
+				s.ADIdentifiers = append(s.ADIdentifiers, container.Name, container.Image)
```

Comment:
`container.Image` has the image tag, right? You should use the same strategy as the docker listener and add the long name without tag, and the short name (without repo prefix), see https://github.com/DataDog/datadog-agent/blob/1f18773ea835b9f9424a95258b87858569fafab9/pkg/collector/listeners/docker.go#L445

I'm not sure `container.Name` should be in first, I'd put it last